src/test/java/org/rulez/demokracia/pdengine/VoteTest.java

```diff
@@ -25,13 +25,28 @@ public void setUp() throws ReportedException {
 	@tested_operation("Cast vote")
 	@tested_behaviour("deletes the ballot with ballotId, so only one vote is possible with a ballot")
 	@Test
-	public void cast_vote_deletes_our_ballot() {
+	public void cast_vote_deletes_ballot_if_canVote_is_true() {
 		String ballot = voteManager.obtainBallot(adminInfo.voteId, adminInfo.adminKey);
 		List<RankedChoice> theCastVote = new ArrayList<RankedChoice>();
 		Vote vote = getTheVote();
+		vote.canVote = true;
 		voteManager.castVote(adminInfo.voteId,ballot,theCastVote);
-		assertFalse(vote.ballots.contains(ballot));
-		
+		assertFalse(vote.ballots.contains(ballot));		
 	}
 
+	@tested_feature("Vote")
+	@tested_operation("Cast vote")
+	@tested_behaviour("works only if canVote is true")
+	@Test
+	public void cast_vote_does_not_delete_ballot_if_canVote_is_false() {
+		String ballot = voteManager.obtainBallot(adminInfo.voteId, adminInfo.adminKey);
+		List<RankedChoice> theCastVote = new ArrayList<RankedChoice>();
+		Vote vote = getTheVote();
+		vote.canVote = false;
+		
+		assertThrows(() -> voteManager.castVote(adminInfo.voteId,ballot,theCastVote)
+				).assertMessageIs("Cast vote does not get recorded");
+		
+		assertTrue(vote.ballots.contains(ballot));
```

Comment:
https://adadocs.demokracia.rulez.org/zenta-tools/magwas/master/zenta-tools/codingrules.html#9cd3f59c-d2f5-465e-a7cc-13eebe455d4c

Reply:
I fixed this, but I do not get your previous comment. 
Could you please explain a little bit more detailed?

Reply:
The functionality is not fully covered yet.
The test for the cast vote not being recorded when canVote is false is nonexisting.
You can either implement is, or explicitly ask to merge the pull request as is, and leave implementation of that functionality to someone else.
Cast vote is different than ballot: ballot is an identifier with which one can cast a vote.
Cast vote is the recorded list of preferences, recorded through the castVote() call.